src/main/java/tc/oc/pgm/match/MatchPlayerImpl.java

```diff
@@ -172,7 +174,13 @@ public boolean canInteract() {
 
   @Override
   public boolean canSee(MatchPlayer other) {
-    return (isObserving() || other.isParticipating()) && other.isVisible();
+    if (!other.isVisible()) return false;
+    if (other.isParticipating()) return true;
+    if (isObserving()) {
+      SettingValue observers = getSettings().getValue(SettingKey.OBSERVERS);
+      return observers.equals(SettingValue.OBSERVERS_SHOWN);
+    }
+    return false;
```

Comment:
This 4th branch should never occur, or am i missing something? If that's the case, maybe leave a comment saying that shouldn't happen.

Reply:
Maybe throw `IllegalStateException`?

Reply:
Maybe add a logger, but don't throw an exception, it's better not to crash PGM because of something that can easily be recoverable by just returning an arbitrary false

Reply:
`canX` methods should never return an illegal state, see my comment below.